src/modules/messaging/index.js

```diff
@@ -79,12 +79,12 @@ export default class Messaging extends ModuleBase {
     return this._ios;
   }
 
-  getToken(): Promise<string> {
-    return getNativeModule(this).getToken();
```

Comment:
I agree with the other remove comment from Salakar - this could be the place where you check for senderID null and use the javascript level token, then we have one set of branching logic in javascript vs two sets, one in each native area. It should serve to simply the native code a bit